app/components/work_package_relations_tab/relation_component.rb

```diff
@@ -1,44 +1,65 @@
+# modify it under the terms of the GNU General Public License version 3.
+#
+# OpenProject is a fork of ChiliProject, which is a fork of Redmine. The copyright follows:
+# Copyright (C) 2006-2013 Jean-Philippe Lang
+# Copyright (C) 2010-2013 the ChiliProject Team
+#
+# This program is free software; you can redistribute it and/or
+# modify it under the terms of the GNU General Public License
+# as published by the Free Software Foundation; either version 2
+# of the License, or (at your option) any later version.
+#
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with this program; if not, write to the Free Software
+# Foundation, Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301, USA.
+#
+# See COPYRIGHT and LICENSE files for more details.
+#++
+
 class WorkPackageRelationsTab::RelationComponent < ApplicationComponent
   include ApplicationHelper
   include OpPrimer::ComponentHelpers
 
-  attr_reader :work_package, :relation, :visibility, :child, :editable
+  attr_reader :relation_item, :editable
+
+  delegate :relation, :work_package, to: :relation_item
 
   # Checks if the relation or child work package is visible to the current user
   #
-  # @param work_package [WorkPackage] The work package whose relations are being displayed
-  # @param relation [Relation, nil] The relation between work packages, if any
-  # @param visibility [Symbol] The visibility status of the relation (:visible or :ghost)
-  # @param child [WorkPackage, nil] The child work package, if this is a parent-child relationship
+  # @param relation_item [WorkPackageRelationsTab::RelationsMediator::RelationItem] The relation item to display
   # @param editable [Boolean] Whether the relation can be edited
-  # @param closest [Boolean] Whether the follows relation is the closest
-  def initialize(work_package:,
-                 relation:,
-                 visibility:,
-                 child: nil,
-                 editable: true,
-                 closest: false)
+  def initialize(relation_item:, editable: true)
     super()
 
-    @work_package = work_package
-    @relation = relation
-    @visibility = visibility
-    @child = child
+    @relation_item = relation_item
     @editable = editable
-    @closest = closest
   end
 
   def related_work_package
-    @related_work_package ||= if parent_child_relationship?
-                                @child
-                              else
-                                relation.from == work_package ? relation.to : relation.from
-                              end
+    relation_item.related
+  end
+
+  def child
+    if parent_child_relationship?
+      relation_item.related
+    end
```

Comment:
```suggestion
   relation_item.related if parent_child_relationship?
```

reads a bit better to me.

Reply:
That's funny. Actually it does not read better for me.

When I scan it, I tend to omit the `if` and I think that it always returns a value. It's only once I read it that I see the `if`. That's why most of the time I reserve one-line if for guard clauses.

What would read better for me would be something like this:

```
return unless parent_child_relationship?

relation_item.related
```

Also, I could inline the `child` and replace its call with `relation_item.related` directly but I found `child` more intention-revealing.

What do you think?

Reply:
I used the code you suggested.